WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -196,6 +196,16 @@ extension PointOfSaleAggregateModel {
         }
     }
 
+    @MainActor
+    func sendReceipt(to emailAddress: String) async {
+        guard let order = orderController.order else {
+            return
+        }
```

Comment:
Could you use orderState here? As per the deprecation note, we'll want to remove the `order` property as it's not the ideal source of truth. It may be that the view is passing the order, because it should (presumably) only be showing the receipt button when there's an order loaded, not when it's syncing or errored?

Reply:
The deprecation notice throw me off a bit here, as I don't see right away how we can use it to retrieve the order, so I though was still ongoing: The `orderState` property is of type `PointOfSaleOrderState`, which in its `.loaded` case only contains several strings we display in the view, and not the `order` object itself, we can get this from `PointOfSaleInternalOrderState` but I didn't wanted to use much combine since we're trying to get away from it (maybe not in this context, since it's internal observation only?)

I've made the changes here: https://github.com/woocommerce/woocommerce-ios/pull/14481/commits/e7e7a76d1523647a3595a352692f1e94439a6b57, let me know if this is what you had in mind, in which case we can change the collectPayment implementation as well on a different PR, or I can revert it and iterate.